examples/download_file.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+require 'uber_task'
+require 'colorize'
+require 'open-uri'
+require 'tmpdir'
+
+VALID_FILE_URI = 'https://github.com/shakacode/uber_task/raw/main/README.md'
+NON_EXESTENT_FILE_URI = 'https://github.com/shakacode/uber_task/raw/main/NON_EXISTENT.md'
+
+module Examples
+  class DownloadFile
+    def self.run(to:, source: VALID_FILE_URI, **task_options)
+      UberTask.run(
+        'Download file from external source',
+        default_retry_wait: 3,
+        retry_count: 5,
+        vital: false,
+        **task_options,
+      ) do
+        UberTask.on_retry do
+          UberTask.logger.info(
+            "Retrying to download file from #{source}...".yellow,
+          )
+        end
+
+        UberTask.on_success do
+          UberTask.logger.info(
+            "Downloaded file #{source} and saved it at #{to}".green,
+          )
+        end
+
+        save_external_file!(source: source, to: to)
+      end
+    end
+
+    def self.save_external_file!(source:, to:)
+      URI.parse(source).open { |io| File.write(to, io.read) }
+    rescue OpenURI::HTTPError => err
```

Comment:
Move this check to the parent task on_subtask_error; all errors bubble up to the parent, like in Javascript and if the parent asks the task to be retried, the current task (child) will run again.

This was the main idea to deploy apps on popmenu, we have a single place for common network errors and tasks can have many subtasks but all of them have common network errors and each time they happen, the on_subtask_error of the top task checks the error type and just ask to retry.

Reply:
Moved `rescue` from here to parent task's `on_subtask_error`